Include/cpython/pylifecycle.h

```diff
@@ -63,14 +63,20 @@ typedef struct {
         .gil = PyInterpreterConfig_OWN_GIL, \
     }
 
+#ifdef Py_GIL_DISABLED
+#  define _PyInterpreterConfig_LEGACY_CHECK_MULTI_INTERP_EXTENSIONS 1
```

Comment:
It would be worth having a note here that this is not intended to be a long-term solution to gh-117649.